src/index.ts

```diff
@@ -100,16 +104,25 @@ export function useSWRNativeRevalidate<Data = any, Error = any>(
     }
 
     let unsubscribeFocus: ReturnType<typeof addListener> | null = null
+    let unsubscribeAppStateChange: AppStateAddEventListenerReturn | null | void = null
 
     if (revalidateOnFocus) {
       unsubscribeFocus = addListener('focus', onFocus)
-      AppState.addEventListener('change', onAppStateChange)
+      unsubscribeAppStateChange = AppState.addEventListener(
+        'change',
+        onAppStateChange
+      )
     }
 
     return () => {
       if (revalidateOnFocus) {
         unsubscribeFocus?.()
-        AppState.removeEventListener('change', onAppStateChange)
+
+        if (unsubscribeAppStateChange && unsubscribeAppStateChange?.remove) {
```

Comment:
I think this should work right?
```ts
 if (unsubscribeAppStateChange?.remove) {
          unsubscribeAppStateChange.remove()
```

Reply:
Yes, it works, I had done it this way initially, however, the typescript showed some errors and did not allow the execution.

Reply:
@nandorojo any positioning?